Formula/git-sync.rb

```diff
@@ -0,0 +1,45 @@
+class GitSync < Formula
+  desc "Clones a git repository and keeps it synchronized with the upstream"
+  homepage "https://github.com/kubernetes/git-sync#readme"
+  url "https://github.com/kubernetes/git-sync/archive/refs/tags/v3.6.0.tar.gz"
+  sha256 "0b65c3f414c98b13d66e99c86680c67348be50d578f88a0888720888b3fcd700"
+  license "Apache-2.0"
+
+  head do
+    url "https://github.com/kubernetes/git-sync.git", branch: "master"
+    depends_on "pandoc" => :build
+  end
+
+  depends_on "go" => :build
+
+  depends_on "coreutils"
+
+  def install
+    ENV["CGO_ENABLED"] = "0"
+    ENV["GOFLAGS"] = "-mod=vendor"
+    inreplace "cmd/git-sync/main.go", "\"mv\", \"-T\"", "\"#{Formula["coreutils"].opt_bin}/gmv\", \"-T\"" if OS.mac?
+    build_flags = "build -v -trimpath -ldflags"
+    build_gmod = "k8s.io/git-sync"
+    build_root = buildpath/"cmd/git-sync"
+    pandoc_opts = "-V title=git-sync -V section=1"
+    cd build_root do
+      system "#{Formula["go"].bin}/go #{build_flags} \"-X #{build_gmod}/pkg/version.VERSION=v#{version}\" ./..."
+      bin.install "git-sync"
```

Comment:
```suggestion
    ldflags = "-X k8s.io/git-sync/pkg/version.VERSION=v#{version}
    system "go", "build", std_go_args(ldflags: ldflags), "./cmd/git-sync"
```

Reply:
Pandoc's options still need to be set but only when generating the man page - I'll adjust that also in a bit.

Reply:
As an aside, it is unfortunate that v4.0 isn't out yet - besides the man page generation and somewhat incompatible but saner command-line options, v4 of this tool has better log output, always properly-formed JSON, so just piping to `jq` works to pretty-print the output, useful when using this tool outside of k8s.

The release notes say that this v3.6.0 will likely be the final version 3 release, and they'll work towards releasing v4 soon.

Reply:
Should be much better, using `*std_go_args` in https://github.com/Homebrew/homebrew-core/pull/106907/commits/386246077b56008ac5b37936dd09c4dd6a8e73ec and also robustly handles changes in import path (via `go list -m`).

Reply:
Rebased and force pushed after adding conflicts to appease the bot.